lib/src/db/db.dart

```diff
@@ -402,14 +414,21 @@ Future _commitHelper(DatastoreDB db,
   if (deletes != null) {
     entityDeletes = deletes.map(db.modelDB.toDatastoreKey).toList();
   }
+  Future<ds.CommitResult> r;
+  if (datastoreTransaction != null) {
+    r = db.datastore.commit(
+        inserts: entityInserts ?? [],
+        autoIdInserts: entityAutoIdInserts ?? [],
+        deletes: entityDeletes ?? [],
+        transaction: datastoreTransaction);
```

Comment:
Nit: same here: why can't we pass `transaction: datastoreTransaction`?

Reply:
because if we allow `transaction: datastoreTransaction` then `transaction` have to be nullable..

And we might prefer that users don't pass a `datastoreTransaction` that they forgot to initialize.